drivers/dma/dma_mcux_lpc.c

```diff
@@ -179,12 +182,53 @@ static int dma_mcux_lpc_configure(const struct device *dev, uint32_t channel,
 
 	LOG_DBG("channel is %d", p_handle->channel);
 
-	if (config->source_chaining_en && config->dest_chaining_en) {
-		LOG_DBG("link dma out 0 to channel %d", config->linked_channel);
-		/* Link DMA_OTRIG 0 to channel */
-		INPUTMUX_AttachSignal(INPUTMUX, 0, config->linked_channel);
+	if ( config->source_chaining_en || config->dest_chaining_en) {
+		if ( !k_sem_take(&is_trig_mux_in_use, K_MSEC(200))) {
+			LOG_DBG("link dma out 0 to channel %d", config->linked_channel);
+			/* At the moment we only have support
+			* for OTrig and are planning to add
+			* support for X amount of OTrigs that
+			* each SOC supports that use DMA_MCUX_LPC
+			*/
+#if CONFIG_SOC_SERIES_IMX_RT5XX
+			/* Link DMA_OTRIG 0 to channel */
+			INPUTMUX_AttachSignal(INPUTMUX, 0,
+					kINPUTMUX_Dma0OtrigChannel0ToTriginChannels + channel);
+			/* Since INPUTMUX handles the signal component
+			* to channel chaining, we should request that
+			* the trigger is handled via INPUTMUX hardware
+			* trigger by default
+			*/
+			DEV_BASE(dev)->CHANNEL[config->linked_channel].CFG |=
+				DMA_CHANNEL_CFG_HWTRIGEN_MASK;
+			DMA_EnableChannel(DEV_BASE(dev), config->linked_channel);
+			INPUTMUX_AttachSignal(INPUTMUX, config->linked_channel,
+					kINPUTMUX_Dma0TrigOutAToDma0);
+#elif CONFIG_SOC_SERIES_LPC55XXX
+			/* Link DMA_OTRIG 0 to channel */
+			INPUTMUX_AttachSignal(INPUTMUX, 0,
+				kINPUTMUX_Dma0Hash0TxTrigoutToTriginChannels
+				+ channel);
+			/* Since INPUTMUX handles the signal component
+			* to channel chaining, we should request that
+			* the trigger is handled via INPUTMUX hardware
+			* trigger by default
+			*/
+			DEV_BASE(dev)->CHANNEL[config->linked_channel].CFG |=
+				DMA_CHANNEL_CFG_HWTRIGEN_MASK;
+			DMA_EnableChannel(DEV_BASE(dev), config->linked_channel);
+			INPUTMUX_AttachSignal(INPUTMUX, config->linked_channel,
+					kINPUTMUX_Otrig0ToDma0 );
+#endif //CONFIG_SOC_MIMXRT595_EVK
+			dma_data->is_inputmux_setup |= 0x1;
+			k_sem_give(&is_trig_mux_in_use);
+		} else { // k_sem_take(..);
+		LOG_ERR("At the moment, this driver can only support 1 trigger"
+		"\n mux and only allows for one channel to trigger "
+		"\n other dma channels\n");
```

Comment:
I prefer if my lines of code don't hit some boundary limit of characters
for at least half the screen, since this is mostly a log message, I imagine it will take 
up about the same amount of space so I added the '\n'. This prevents, in such situations (mainly chained error messages)
from having to wrap text around.

Reply:
this will mess up log files, there should be one log per line